app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepositoryImpl.kt

```diff
@@ -98,63 +99,63 @@ class AgendaRepositoryImpl @Inject constructor(
 //                    }
 //                }
 
-                withContext(Dispatchers.IO) {
-
-                    // • Events - Insert fresh data locally in parallel
-                    eventRepository.clearEventsForDayLocally(dateTime)
-                    agenda?.events?.map { event ->
-                        supervisorScope {
-                            async {
-                                val upsertResult =
-                                    eventRepository.upsertEventLocally(event.toDomain())
-                                if (upsertResult is ResultUiText.Error) {
-                                    throw IllegalStateException(upsertResult.message.asStrOrNull())
+                    return withContext(Dispatchers.IO) {
+
+                        // • Events - Insert fresh data locally in parallel
+                        eventRepository.clearEventsForDayLocally(dateTime)
+                        agenda?.events?.map { event ->
+                            supervisorScope {
```

Comment:
You are creating a supervisorScope on each item from the event. You can run the supervisorScope outside:

```
supervisorScope {
  agenda?.events?.map { event -> 
  // etc
  }
```

Reply:
OK will refactor.